src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,111 @@
+are first downloaded from the bucket and put in the cache before a filehandle is made to access
+them.
+
+The storage source uses the filesystem's put-object functionality to flush the files into the cloud.
+The Tier Manager inside WiredTiger directs the flush of the files. Only S3 storage source files
+pushed to the cloud are also copied to the local cache for easy access.
+
+@section extension_cloud_file_handle Cloud File Handle Interface
+
+Each storage source also implements the \c WT_FILE_HANDLE interface, to access the files
+on the file system. Since the object store is read-only, all the write access methods are disabled.
+
+@section extension_log_stat Messaging and Statistics
+The cloud provider SDKs uses the terms log systems and logging levels differently to WiredTiger.
+The SDK's logging definition refers to the SDK messages that are generated and the logging levels
+control the amount of information given to the user. The SDKs provide their own levels and
+are translated into WiredTiger messaging levels.
+
+All of the cloud storage sources have their own messaging system and statistics implemented as a
+method to monitor the state of the SDKs and the storage sources. The messaging system captures both
+SDK and the custom storage source error and informational messages, and redirect it into the
+WiredTiger messaging system. The statistics are also printed when the \c WT_VERBOSE level is set on the
+storage source via the custom messaging system.
+
+@section s3_store S3 Storage source
+The S3 store implements the following key functionalities defined by the storage source interface:
+
+- The bucket is expected to be a string argument, which is the bucket name and the
+corresponding region separated by a ';', e.g. "bucket1;ap-southeast-2". AWS requires an access token
+to authenticate programmatic access to S3. The access token comprises an access key ID and a
+secret access key as a set. The S3 store requires the token to be provided as a single string
+argument of the access key followed by the secret key, separated by a ';'.
+
+- The S3 store also configures a directory on the file system local to the WiredTiger database as a
+cache for the object files. The cache is used to improve the read latency for the object files read
+after an initial download from the S3.
+
+- When a S3 file handle is created, the local cache gets a copy of the S3 object if
+it doesn't already have it. Since the local cache is a directory on the WiredTiger database's file
+system, the file is accessed using a \c WT_FILE_HANDLE implementation native to WiredTiger.
+Hence, a copy of native \c WT_FILE_HANDLE is kept encapsulated in the S3 file handle.
+
+- The S3 store has created a logging class which extends the AWS SDK's logging system.
+
+@section azure_gcp_store Azure and GCP Storage Source
+The Azure and GCP store implements the following key functionalities differently from S3:
```

Comment:
Grammar: "The Azure and GCP store implements the following..." should be "The Azure and GCP stores implement the following"